packages/datetime/src/timePicker.tsx

```diff
@@ -61,6 +61,17 @@ export interface ITimePickerProps extends IProps {
      */
     disabled?: boolean;
 
+    /**
+     * Props passed through to each rendered input element.
+     */
+    inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+
+    /**
+     * Custom input component that will be used instead of a pure input element.
+     * Needs to have the same interface as a normal input.
+     */
+    inputComponent?: React.ComponentType<React.InputHTMLAttributes<HTMLInputElement>>;
```

Comment:
usually we make a "renderX" prop instead of asking for a component... so how about we change this to `renderInput: (props: React.InputHTMLAttributes<HTMLInputElement>) => JSX.Element`

Reply:
of course we can do it like this, although i believe this is a different (and more far reaching) change.

the way i would expect a passable render function to behave is different from a simple component. More specifically, i would expect a render function to receive all relevant props and states that get handled by the default  method and hand off implementation completely to the user. My solution was a bit smaller in scope.

So the interface for a `renderInput` would look like this:
```tsx
const renderInputProps = {
  disabled: boolean;
  inputProps: React.InputHTMLAttributes<HTMLInputElement>;
  value: string;
  onBlur: (e: React.SyntheticEvent<HTMLInputElement>) => void,
  onChange: (e: React.SyntheticEvent<HTMLInputElement>) => void,
  onFocus: (e: React.SyntheticEvent<HTMLInputElement>) => void,
  onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void,
}
// ...
renderInput:  (renderProps: renderInputProps) => React.ComponentType
```
implementation would then be in the `render` method, where we either call `this.props.renderInput(/* ... */)` or fall back to `this.renderInput(/* ... */)`. 

While providing more control, this definitely complicates usage and as such i would not advocate for it. However, anything other than a complete replacement for the actual `renderInput` method of the component would be confusing.